### PR TITLE
Addition of a field to allow adding a personalized url. Because endpo…

### PR DESCRIPTION
**Pull request by camilleislasse**
Addition of SogeCommerce Endpoint ( Societé Génerale )

Addition of a field to allow adding a personalized url. Because endpoints can change and it's restrictive to have endpoints hard-coded in the code

### CODE REVIEW

**Comment by camilleislasse** [comment]
@ekyna

**Comment by Prometee** [comment]
Wonderful @camilleislasse !
Maybe adding a test for the `endpoint_url` could be great 😉

File: src/Api/Api.php [hunk 1]
```diff
@@ -479,6 +482,10 @@ private function getHashModes(): array
 
     private function getUrl(): string
     {
+
```
Comment (ekyna) [inline src/Api/Api.php]: Please remove this blank line.

File: src/PayzenGatewayFactory.php [hunk 2]
```diff
@@ -53,6 +53,7 @@ protected function populateConfig(ArrayObject $config)
                 'ctx_mode'    => null,
                 'directory'   => null,
                 'endpoint'    => null,
+                'endpoint_url'    => null,
```
Comment (ekyna) [inline src/PayzenGatewayFactory.php]: Please align arrow.

File: src/PayzenGatewayFactory.php [hunk 3]
```diff
@@ -66,6 +67,7 @@ protected function populateConfig(ArrayObject $config)
 
                 $payzenConfig = [
                     'endpoint'    => $config['endpoint'],
+                    'endpoint_url'=> $config['endpoint_url'],
```
Comment (ekyna) [inline src/PayzenGatewayFactory.php]: There should be a space between string and arrow. Then align arrows.

**Comment by ekyna** [comment]
As you are using PHPStorm, you should configure PSR as coding style.
Create a new Scheme > Click « Set from... » > Select « PSR12 ».
image
Then you can use `Ctrl + Alt + L` to reformat code.

**Comment by ekyna** [comment]
Thx @camilleislasse

**Pull request merged**